### PR TITLE
add page for each category, add burger in header component.

### PR DESCRIPTION
**Pull request by Irazhap**
delete movielist and create carousel component

### CODE REVIEW

File: components/Card.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default function Card({ image, title, year }) {
     <>
       <CardImage src={image} alt={title} width={180} height={250} />
```
Comment (aaaleeenaaa) [inline components/Card.js:23]: This is not directly related to the user story, but I just realized that you add other width and height here than in the styled CardImage component. Is that intended?

File: components/Header.js [hunk 2]
```diff
@@ -1,23 +1,111 @@
...
+          <NavbarLink href="/index">Home</NavbarLink>
+          <NavbarLink href="/popular">Popular</NavbarLink>
+          <NavbarLink href="/toprated">Top Rated</NavbarLink>
+          <NavbarLink href="/upcoming">Upcoming</NavbarLink>
+        </NavbarMenuItem>
+      </NavbarMenu>
+      <NavUnderContainer isOpen={isOpen} />
       <HeaderTitle>The Movie App</HeaderTitle>
     </NavContainer>
   );
```
Comment (aaaleeenaaa) [inline components/Header.js:111]: There are a lot of styled components for the navigation, you might want to think about outsourcing them? Or at least giving the component a more understandable name (like "navigation" or something like that), as for me it is not really a header

File: components/Header.js [hunk 3]
```diff
@@ -1,23 +1,111 @@
...
+  height: 100%;
+  background: #000;
+  opacity: 0.5;
+  z-index: 7777;
+  transition: transform 0.3s ease-in-out;
+  transform: ${({ isOpen }) =>
+    isOpen ? "translateX(0)" : "translateX(-100%)"};
+`;
+
+const NavbarMenuItem = styled.ul`
```
Comment (aaaleeenaaa) [inline components/Header.js:50]: I think the name "NavbarMenuItem" could be misunderstood. For me, it would be a list item and not the list itself, so you might think about renaming it

File: components/Header.js [hunk 4]
```diff
@@ -1,23 +1,111 @@
 import styled from "styled-components";
+import Link from "next/link";
 
 const NavContainer = styled.header`
   display: flex;
+  position: fixed;
   justify-content: center;
   align-items: center;
+  top: 0px;
   height: 60px;
```
Comment (aaaleeenaaa) [inline components/Header.js:10]: Just a general note, not just for this but for everything: You might want to think about using relative units (like rem etc.) so that the sizes can adjust

**Comment by Irazhap** [comment]
Hi Alena. Thank you for your comments, I have implemented many of them as possible. I'll make the adjustments with the "em" and "rem" for later. LGTM

File: components/StyledNavContainer.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  width: 250px;
+  height: 100%;
+  background: #f2f2f2;
+  z-index: 8888;
+  transition: transform 0.3s ease-in-out;
+  transform: ${({ isOpen }) =>
+    isOpen ? "translateX(0%)" : "translateX(-100%)"};
+`;
+
+export const StyledTransparentContainer = styled.nav`
```
[the remark this answers is not in the document]
Comment (Irazhap) [reply components/StyledNavContainer.js:16]: Hi Ralf, thank you for your comments
At the moment i don't have a single approach how to implement this. For my purposes, it works at the moment and I would like to keep it that way, because I'm already sitting on this UserStory for 3 days and still have a lot of work ahead of me....

[the remark this answers is not in the document]
Comment (Irazhap) [reply components/StyledNavContainer.js:16]: @actyralf 
Hi Ralf, thank you for your comments
At the moment i don't have a single approach how to implement this. For my purposes, it works at the moment and I would like to keep it that way, because I'm already sitting on this UserStory for 3 days and still have a lot of work ahead of me....

[the remark this answers is not in the document]
Comment (mbosselmann) [reply components/StyledNavContainer.js:16]: I think Ralf meant that you can use props do adapt your styled components. [You can check the docs (for later).](https://neuefische-students.slack.com/archives/C04RE009UTS/p1683786453421679)

**Comment by mbosselmann** [comment]
Hi Irazhap,
LGTM! 🦊

File: components/Header.js [hunk 6]
```diff
@@ -1,24 +1,27 @@
...
+      <Burger isOpen={isOpen} onClick={onToggle} />
+      <StyledNavContainer isOpen={isOpen}>
+        <StyledNavList>
+          <StyledListItem href="/index">Home</StyledListItem>
+          <StyledListItem href="/popular">Popular</StyledListItem>
+          <StyledListItem href="/toprated">Top Rated</StyledListItem>
+          <StyledListItem href="/upcoming">Upcoming</StyledListItem>
+        </StyledNavList>
+      </StyledNavContainer>
+      <StyledTransparentContainer isOpen={isOpen} />
```
Comment (Roland-Hufnagel) [inline components/Header.js:23]: Don't know what this StyledTransparentContainer is for? Now you have two `nav`elements on your site.

File: components/Header.js [hunk 7]
```diff
@@ -1,24 +1,27 @@
...
-`;
-
-export default function Header() {
+export default function Header({ isOpen, onToggle }) {
   return (
-    <NavContainer>
-      <HeaderTitle>The Movie App</HeaderTitle>
-    </NavContainer>
+    <StyledHead>
+      <Burger isOpen={isOpen} onClick={onToggle} />
```
Comment (Roland-Hufnagel) [inline components/Header.js:14]: Since the `Burger` is just a `div`, he doesn't expect an 'isOpen'-prop. Please delete it here.

File: components/Burger.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  div {
+    width: 100%;
+    height: 4px;
+    background-color: #f2f2f2;
+  }
+`;
+
+export default function Burger({ onClick }) {
```
Comment (Roland-Hufnagel) [inline components/Burger.js:27]: Since only buttons should have an 'onClick' event, you should not use a `styled.div` but a `styled.button`

File: pages/_app.js [hunk 9]
```diff
@@ -1,9 +1,16 @@
...
+import { useState } from "react";
 
 export default function App({ Component, pageProps }) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleToggle = () => setIsOpen(!isOpen);
+
   return (
     <>
+      <Header isOpen={isOpen} onToggle={handleToggle} />
```
Comment (Roland-Hufnagel) [inline pages/_app.js:13]: I think since you now have your header component in your Layout component you don't need it here anymore, right?

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Hi Irazap, some hints in general: For your burgermenu I recommend to create a Navigation-component that is either rendered (depending on a useState, like you did) or not. Then you can handle all the navigation stuff in that component. 

I added some more comments. Please have a look at it.
